background.ts

```diff
@@ -324,6 +294,61 @@ Electron.app.whenReady().then(async() => {
   }
 });
 
+/**
+ * Checks for the existence of the 'backend.lock' file and updates the
+ * 'backendIsLocked' variable accordingly. Emits the 'backend-locked-update'
+ * event to notify listeners about the current lock status.
+ *
+ * This function blocks execution until the 'backend.lock' file is no longer
+ * present.
+ */
+function checkForBackendLock() {
+  return new Promise((resolve) => {
+    const intervalId = setInterval(() => {
+      if (!fs.existsSync(path.join(paths.appHome, 'backend.lock'))) {
```

Comment:
Since this is in an sync function, we should be able to use the promise version here.

```ts
  const intervalId = setIntervnal(async() => {
    try {
      await fs.promises.access(path.join(…));
      backendIsLocked = SNAPSHOT_OPERATION;
      mainEvents.emit(…);
    } catch (ex) {
      if (ex.code === 'ENOTEXIST') {
        throw ex;
      }
      backendIsLocked = '';
      …
```

Reply:
This is a good point.. I can't say where my head was at with this one other than that the original implementation utilized `existsSync`..

Reply:
I ended doing more refactoring than I anticipated..

It felt more natural to separate this into two functions: one that checks for the existence of a backend lock, and the  other that handles the flow of execution. Let me know what you think

Reply:
I believe the original implementation used `existsSync` because there's no `fs.promises.exists` and `fs.exists` (the callback version) is deprecated.

(I haven't looked at the code yet, but having a separate function to check for the existence of a backend lock sounds great)